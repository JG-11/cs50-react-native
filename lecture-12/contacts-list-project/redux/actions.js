import { retrieveQuote } from './../API'

export const UPDATE_USER = 'UPDATE_USER'
export const UPDATE_CONTACT = 'UPDATE_CONTACT'
export const GET_QUOTE = 'GET_QUOTE'

export const updateUser = update => ({
    type: UPDATE_USER,
    payload: update
})
export const addContact = newContact => ({
    type: UPDATE_CONTACT,
    payload: newContact
})
export const getQuote = () => async dispatch => {
    const quote = await retrieveQuote()

    dispatch({ type: GET_QUOTE, payload: quote })
}
