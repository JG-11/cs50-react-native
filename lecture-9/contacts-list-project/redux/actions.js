export const UPDATE_USER = 'UPDATE_USER'
export const UPDATE_CONTACT = 'UPDATE_CONTACT'

export const updateUser = update => ({
    type: UPDATE_USER,
    payload: update
})
export const addContact = newContact => ({
    type: UPDATE_CONTACT,
    payload: newContact
})