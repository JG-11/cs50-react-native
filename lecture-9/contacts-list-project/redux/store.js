import { createStore } from 'redux'
import reducer from './reducer'
import { addContact } from './actions'

const store = createStore(reducer)

/*
store.dispatch(updateUser({foo: 'foo'}))
store.dispatch(updateUser({bar: 'bar'}))
*/

store.dispatch(addContact({name: 'Genaro', phone: '12121212'}))
store.dispatch(addContact({name: 'Leslie', phone: '1212323'}))
store.dispatch(addContact({name: 'Genaro', phone: '121212'}))

export default store