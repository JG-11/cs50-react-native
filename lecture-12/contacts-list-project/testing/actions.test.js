import * as actions from './../redux/actions'

test('updateUser returns an action', () => {
  expect(actions.updateUser({ name: 'test name' })).toMatchSnapshot()
})
