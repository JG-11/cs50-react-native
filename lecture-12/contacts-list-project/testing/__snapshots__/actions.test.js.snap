// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateUser returns an action 1`] = `
Object {
  "payload": Object {
    "name": "test name",
  },
  "type": "UPDATE_USER",
}
`;
